src/providers/mbed_provider/mod.rs

```diff
@@ -107,6 +101,11 @@ impl MbedProvider {
             // Delete those who are not present and add to the local_store the ones present.
             match store_handle.get_all(ProviderID::MbedCrypto) {
                 Ok(key_triples) => {
+                    if let Err(error) = psa_crypto::init() {
+                        error!("Error {} when initialising Mbed Crypto library.", error);
+                        return None;
+                    }
```

Comment:
What's the purpose of this init if the one on line 77 happens?

Reply:
I'm not sure why or how that got there! It definitely shouldn't be.